src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -52,67 +76,165 @@ public void AddDependencyLocation(string fullPath)
                 {
                     _knownAssemblyPathsBySimpleName[simpleName] = paths.Add(fullPath);
                 }
+
+                _ = _analyzerAssemblyInfoMap[fullPath] = null;
```

Comment:
A comment is probably warranted here. Also, the discard assignment seems unnecessary.

I am guessing that when we add a dependency location, we want to "reset" the process of calculating a shadow path for it. Is that right?

Reply:
Correct. 

The scenario is the following: 

1. Analyzer.csproj builds to `bin\debug\analyzer.dll` 
2. Consumer.csproj leverages `analyzer.dll`. 
3. The solution is built multiple times

Every time we essentially get a new call to `AddDependencyLocation`. The second call needs to supercede the first one else the second build will re-load the first version of `analyzer.dll`